dist/backdraft.js

```diff
@@ -1303,6 +1303,19 @@ _.extend(Plugin.factory, {
       }
     },
 
+    // takes a hash of { columnTitle: columnState, ... }
+    setColumnVisibilities: function(columns) {
+      _.each(columns, function(state, title) {
+        if (!state && this._columnManager.columnConfigForTitle(title).required) {
+          throw new Error("can not disable visibility when column is required");
+        }
+      }, this);
+      this._columnManager.visibility.set(columns);
```

Comment:
but there's also `changeColumnVisibilities` in web side, isn't that turning off the "saveOnVisibilityChanges", then invoking this code, then turning it back on?  And if not, what is `this._settingsManager.saveOnVisibilityChanges = saveVisibility || userInitiated;` line really doing..